wagtail/models/__init__.py

```diff
@@ -798,6 +821,37 @@ def default_preview_mode(self):
         """
         return self.preview_modes[0][0]
 
+    @property
+    def preview_sizes(self):
+        """
+        Returns a list of dictionaries, each representing a preview size option for this object.
+        Override this property to customize the preview sizes.
+
+        Each dictionary in the list should include the following keys:
+        - `name`: A string representing the internal name of the preview size.
+        - `icon`: A string specifying the icon's name for the preview size button.
+        - `device_width`: An integer indicating the device's width in pixels.
+        - `default_size`: A boolean, set to `True` for the default preview size.
```

Comment:
I quite like the approach, although I think similar to `preview_modes`, it's simpler to just pick the first element in the list as the default size, so you don't have to specify `default_size` everywhere.

Also, @SebCorbin 

> Do you think it would be possible to also configure preview sizes depending on preview mode? Use case: a "Thank you e-mail" preview mode that is best viewed in tablet size, and mobile and desktop should not be displayed.

This would mean the preview size buttons need to be re-rendered when you change preview modes. While it's certainly possible, I think that requires more complexity that I'm not sure is worth it. For your example use case, I don't see a reason why we shouldn't display the mobile and desktop previews. Surely people read emails on mobile and desktop too? 🤔

Which, by the way, rather than _limiting_ the preview size options, I'm more in favour towards a feature that allows you to customise the preview size directly from the panel, à la DevTools' responsive design mode. That seems more like a power user feature, though...

Reply:
I agree, my use case is a bit too precise, anyway I managed to hack it through `jQuery` events, so I'll monitor the Stimulus PR to ensure it's still doable after.